include/mqt-core/ir/operations/OpType.hpp

```diff
@@ -8,165 +8,37 @@
 #include <unordered_map>
 
 namespace qc {
+
+enum OpTypeFlags {
+  OP_TYPE_INV = 1,
+  OP_TYPE_DIAG = 2,
+};
```

Comment:
`enum class`es are even more strict with conversions and manipulations, it would keep a normal `enum`s and use casts to please clang tidy. Adding a flag `NONE` is indeed nice!